src/server/service/GameService.ts

```diff
@@ -0,0 +1,229 @@
+    private playerRepository = PlayerRepository.getInstance();
+    private roomRepository = RoomRepository.getInstance();
+
+    private constructor() { }
+
+    public static getInstance(): GameService {
+        if (!GameService.instance) {
+            GameService.instance = new GameService();
+        }
+        return GameService.instance;
+    }
+
+
+    public registerPlayer(clientIndex: string, playerRequest: PlayerRequest): Player {
+        return this.playerRepository.createPlayerOrLogin(playerRequest.name, playerRequest.password, clientIndex);
+    }
+
+    public createRoom(playerIndex: string): Room {
+        const createdRoom = this.roomRepository.createRoom();
+        try {
+            const currentPlayer = this.playerRepository.getPlayerByIndex(playerIndex);
+            if (!currentPlayer) throw new Error('Player not found');
+            this.roomRepository.addUserToRoom(createdRoom.indexRoom, currentPlayer);
+        }
+        catch {
+            this.checkGameFinishOrEnd(createdRoom);
+        }
+        return createdRoom;
+    }
+
+    public addUserToRoom(playerIndex: string, indexRoom: string) {
+        const currentPlayer = this.playerRepository.getPlayerByIndex(playerIndex);
+        if (!currentPlayer) throw new Error('Player not found');
+        
+        this.roomRepository.addUserToRoom(indexRoom, currentPlayer);
+    }
+
+    public updateRoom(roomIndex: string): Room {
+        const room = this.roomRepository.getRoomByIndex(roomIndex);
+        if (!room) throw new Error('Room not found');
+
+        if (room.players.length >= 2) {
+            return room;
+        } else if (room.players.length === 0) {
+            this.roomRepository.removeRoom(room);
+        }
+
+        throw new Error('Room state undefined');
+    }
+
+    public getRoom(roomIndex: string): Room | undefined {
+        return this.roomRepository.getRoomByIndex(roomIndex);
+    }
+
+    public getFreeRooms(): Room[] {
+        return this.roomRepository.getAllRooms().filter(room => room.players.length < 2);
+    }
+
+    public addShipsToRoom(playerIndex: string, shipsData: AddShipsRequest): boolean {
+        const room = this.roomRepository.getRoomByIndex(shipsData.gameId);
+        if (!room || room.isStarted) throw new Error('Room not found or battle already started');
+
+        const ships: Ship[] = shipsData.ships.map(ship => ({
+            position: ship.position,
+            direction: ship.direction,
+            length: ship.length,
+            type: ship.type,
+            hits: 0
+        }));
+        room.ships.set(playerIndex, ships);
+
+        return room.ships.size >= 2;
+    }
+
+    public startGame(roomIndex: string): Room {
+        const room = this.roomRepository.getRoomByIndex(roomIndex);
+        if (!room || room.isStarted) throw new Error('Room not found or battle already started');
+
+        room.isStarted = true;
+        this.switchToNextPlayer(room);
+        return room;
+    }
```

Comment:
You need to clean up any room that has been created by any of the users in the room.